lib/content-services/directives/node-lock.directive.ts

```diff
@@ -0,0 +1,53 @@
+/*!
+ * @license
+ * Copyright 2016 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* tslint:disable:no-input-rename  */
+
+import { Directive, ElementRef, Renderer2, HostListener, Input, Output, EventEmitter, AfterViewInit } from '@angular/core';
+import { MinimalNodeEntryEntity } from 'alfresco-js-api';
+import { PermissionsEnum, ContentService } from '@alfresco/adf-core';
+import { ContentNodeDialogService } from '../content-node-selector/content-node-dialog.service';
+
+@Directive({
+    selector: '[adf-node-lock]'
+})
+export class NodeLockDirective implements AfterViewInit {
+
+    @Input('adf-node-lock')
+    node: MinimalNodeEntryEntity;
+
+    @Output()
+    error: EventEmitter<any> = new EventEmitter<any>();
```

Comment:
The error output here is neither documented, nor used in this file. The contentNodeDialogService.openLockNodeDialog should somehow use.